Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -1774,9 +1778,23 @@ public JSONObject getFirstReferringParams() {
         return firstReferringParams;
     }
 
+    private CountDownLatch getFirstReferringParamsLatch = null;
+    public JSONObject getFirstReferringParamsSync() {
+        getFirstReferringParamsLatch = new CountDownLatch(1);
+        try {
+            if (initState_ != SESSION_STATE.INITIALISED) getFirstReferringParamsLatch.await(LATCH_WAIT_UNTIL, TimeUnit.MILLISECONDS);
```

Comment:
In the `FirstReferringParams` we don't need to wait until `INITIALISED ` unless it is a first launch. Can you please  check if the  `FirstReferringParams` available instead of init finished or not in this case